core/src/main/java/org/apache/iceberg/SerializedTable.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.hadoop.SerializedConfiguration;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.apache.iceberg.util.SerializableMap;
+
+/**
+ * A read-only serializable table implementation that can be sent to other nodes.
+ * <p>
+ * While this class captures the metadata file location that can be used to load the full table
+ * metadata, it also serializes directly things like properties, schema, spec, sort order
+ * to avoid reading the metadata file from other nodes to access frequently needed metadata.
```

Comment:
Because `FileIO` and `LocationProvider` can all be dynamically loaded, and user can also potentially have a custom `EncryptionManager`, we should still mention in documentation that the end user needs to ensure Kryo compatibility in case they use custom implementations.

Reply:
Yeah, let me adapt the old message slightly and add it back.

Reply:
Added the old comment back.